tools/checklocks/state.go

```diff
@@ -187,6 +187,35 @@ func (l *lockState) isCompatible(other *lockState) bool {
 	return l.isSubset(other) && other.isSubset(l)
 }
 
+// join returns a new lockState of the intersection of 2 states.
+func (l *lockState) join(other *lockState) *lockState {
+	rls := l.fork()
+
+	for addr, val := range rls.stored {
+		if oval, ok := other.stored[addr]; !ok || val != oval {
+			rls.modify()
+			delete(rls.stored, addr)
+		}
+	}
+
+	if l.isCompatible(other) {
+		return rls
+	}
+
+	for _, m := range rls.lockedMutexes {
+		intersection := make([]string, 0)
```

Comment:
In Go usually a nil slice is not distinguished from an empty but allocated slice. Is there a reason to have this be an allocated slice?

Reply:
There is no reason for this to be an allocated slice, I was unaware that `nil` slices are not distinguish from empty allocated slices

Reply:
One can distinguish (s==nil is not the same as len(s)==0). But it is not usually necessary.